src/lib/ui/ExtensionsList.svelte

```diff
@@ -1,13 +1,54 @@
 <script lang="ts">
-import type { ExtensionByCategoryInfo } from '$lib/api/extensions-info';
+import type { CatalogExtensionInfo, ExtensionByCategoryInfo } from '$lib/api/extensions-info';
 
+import ExtensionByCategoryCard from './ExtensionByCategoryCard.svelte';
 import ExtensionsByCategory from './ExtensionsByCategory.svelte';
 
-const { extensionsByCategories }: { extensionsByCategories: ExtensionByCategoryInfo[] } = $props();
+export let extensionsByCategories: ExtensionByCategoryInfo[];
+
+let filteredExtensions: CatalogExtensionInfo[] = [];
+let showCategories = true;
+
+// Get the query from the URL
+function getQueryLimit(): number | null {
+  const query = new URLSearchParams(window.location.search).get('query');
+  return query ? parseInt(query, 10) : null;
+}
+
+// Sort extensions based upon the last updated date based upon what's in .versions array
+function getSortedExtensions(extensions: CatalogExtensionInfo[]): CatalogExtensionInfo[] {
+  return [...extensions].sort((a, b) => {
+    const aLastUpdated = a.versions?.[a.versions.length - 1]?.lastUpdated ?? 0;
+    const bLastUpdated = b.versions?.[b.versions.length - 1]?.lastUpdated ?? 0;
+    return Number(bLastUpdated) - Number(aLastUpdated);
+  });
+}
+
+$: {
+  const limit = getQueryLimit();
+  showCategories = !limit;
+
+  if (limit) {
+    const allExtensions = extensionsByCategories.flatMap(({ extensions }) => extensions);
+    filteredExtensions = getSortedExtensions(allExtensions).slice(0, limit);
+  }
+}
 </script>
 
 <div class="flex flex-col h-full">
-	{#each extensionsByCategories as extensionByCategoryInfo}
-		<ExtensionsByCategory {extensionByCategoryInfo} />
-	{/each}
+	{#if showCategories}
+		{#each extensionsByCategories as extensionByCategoryInfo}
```

Comment:
issue (non blocking) but not sure why: looks like we have existing tab characters

Reply:
hmm, looks like `pnpm format:fix`does not actually do svelte files too